lib/index.js

```diff
@@ -1097,23 +1131,19 @@ class Hatch {
     // XXX: This allows us to pass params from SOMA worker
     //      directly into libingester with minimal changes
     //      to the actual ingester.
-    // TODO: Do this generic when we have more args
-    _parse_argv(argv) {
-        let options = {};
-        if (argv instanceof Array) {
-            const path_index = argv.indexOf('--path');
-            if (path_index >= 0 &&
-                path_index < argv.length + 1) {
-                options.path = argv[path_index + 1];
-            }
+    _parse_options(options) {
+        options = options || {};
 
-            const no_tgz = argv.indexOf('--no-tgz');
-            if (no_tgz >= 0) {
-                options.no_tgz = true;
-            }
+        const argv_options = options.argv || process.argv.slice(2);
+
+        const config_options = argv(ARGV_CONFIGFILE_OPTIONS, argv_options);
+        if (config_options['config-file']) {
+            const cf = config_options['config-file'];
+            const file_options = JSON.parse(fs.readFileSync(cf, 'utf8'));
+            options = Object.assign(options, file_options);
         }
 
-        return options;
+        return Object.assign(options, argv(ARGV_OPTIONS, argv_options))
```

Comment:
You could easily have only one `ARGV_OPTIONS` variable with `config-file` in it, use it in line 1139, and here do:

```
return Object.assign(file_options, config_options, options);
```

(arguments on the right have priority)